### DISCUSSION TITLE
Can this be used to sync a subset of files to a team vault?

### DISCUSSION THREAD

**Discussion by ArneNostitz** [ObsidianPublisher/obsidian-github-publisher#79] [General]
hi, i am super excited about finding this plugin. I am looking for a way to sync a subset of files (all of these have a frontmatter key to a project) to a git/vault/folder to be shared with my team members. 

best would be if some of these files could be read only, some editable. but in any case they should stay in sync for the team - and if they are editable also be synced back to me. 

i use git-sync for my whole vault to have it on my phone and ipad as well. but having a subset of files (which are scattered all over the folder structure of my vault) to be synced would be great. wondering if this could be done with you plugin? 

thanks!!

**Comment by Mara-Li** [comment]
Technically, it can be possible to do that with the plugin, yes.
Dunno how Obsidian will handle the conversion, but you can test some death&retry things to get what must be removed or activated. 

I think about the links, and the image, if you want to sync the file in multiple direction.

**Reply by ArneNostitz** [reply]
how would that multi directional sync work - eg. when someone edits the file I originally uploaded and uploads it to the git ... can i down-sync as well with this? or is it only uploading to git?

i thought "refresh" is doing that - but that is actually synching local changes to the git.

**Reply by Mara-Li** [reply]
Yeah it's only uploaded on git, so at last you need to pull the edit

**Reply by ArneNostitz** [reply]
any chance of getting a pull for shared files implemented? 😬 as the structure is not the same on git, it seems complicated to do a real bidirectional sync ...

**Reply by Mara-Li** [reply]
the plugin is not for that, so I'm afraid to no :/

The best idea is to get around the github sync plugin and maybe try with gitignore file

**Reply by ArneNostitz** [reply]
i will check that out if its possible to have a second sync there ... thanks! but i will probably use your great tool too! thanks for the quick responses! cheers

**Comment by ArneNostitz** [comment]
hi - its been a long time - but im circling back to this ... how are you doing the single file commit to the repo? this is the only thing i miss in the git sync plugin ... maybe its "easy" to make a "frankenstein" version of both plugins ... your "add this one file to the repo" and gits "sync" (eg pull) capabilities... 

that would be really awesome as it would enable team collaboration in big vaults that are also filled with private info without it sharing it all ...

**Reply by ArneNostitz** [reply]
(in this case it would be best if the plugin would not convert the file in any way but leave it as is btw)

**Reply by Mara-Li** [reply]
No git pull but some frontmatter option can disable all conversion option of the plugin ; so yes, it can be possible

**Reply by ArneNostitz** [reply]
and how do you do the magic of "frontmatter enables file to be uploaded"?

**Reply by Mara-Li** [reply]
well... Share true and use the command to push only one file at time?

**Reply by ArneNostitz** [reply]
yes sure i know that - but can you (as long as it's not too complicated) share what happens "under the hood"? how do you just commit files with the Frontmatter in it - and ignore all the rest?

**Reply by Mara-Li** [reply]
I don't understand what do you mean ;;

**Reply by ArneNostitz** [reply]
the GitHub sync plugin syncs everything. and your plugin has the ability to define files to be pushed by Frontmatter. 

i am trying to understand if it's possible to combine the code of both plugins to create a selective GitHub sync plugin. 

so i was wondering if you could point to the functions in your code that do the file "separation"

:-) but probably it's too high for me anyway.
